### PR TITLE
Add StringUtil.JoinString helper method

### PR DESCRIPTION
**Pull request by pepone**
This small PR adds `StringUtil.JoinString` helper method to join a string an escape it to be compatible with `GetPropertyAsList`

### CODE REVIEW

File: csharp/src/Ice/StringUtil.cs [hunk 1]
```diff
@@ -602,13 +602,64 @@ public static string UnescapeString(string s, int start, int end, string special
             return l.ToArray();
         }
 
+        public static string JoinString(string[] values, string delimiters)
```
Comment (externl) [inline csharp/src/Ice/StringUtil.cs:605]: I think this method needs some documentation. I find it kinda confusing. Maybe we need 3 params?

Since this is really about properties should the name be something like `JoinStringProperty` or something?

File: csharp/src/Ice/StringUtil.cs [hunk 2]
```diff
@@ -9,21 +9,21 @@
...
-        public static int FindFirstOf(string str, string match) => FindFirstOf(str, match, 0);
+        internal static int FindFirstOf(string str, string match) => FindFirstOf(str, match, 0);
 
         //
         // Return the index of the first character in str to
         // appear in match, starting from start. Returns -1 if none is
         // found.
         //
-        public static int FindFirstOf(string str, string match, int start)
+        internal static int FindFirstOf(string str, string match, int start)
```
Comment (externl) [inline csharp/src/Ice/StringUtil.cs:26]: I known it's not part of this PR but are these really useful. Doesn't C# have an `indexOf` method on `string`.

File: csharp/src/Ice/StringUtil.cs (hunk 1 above)
Comment (pepone) [reply csharp/src/Ice/StringUtil.cs:605]: `JoinStringProperty` sounds fine, maybe I should just hardcode the delimiters used by GetPropertyAsList and have a single parameter `values`?

Comment (externl) [reply csharp/src/Ice/StringUtil.cs:605]: I also think we could just hardcode the "property delimiters" `, \t\r\n`

Comment (externl) [reply csharp/src/Ice/StringUtil.cs:605]: Yes, I think we can "hard code" all of this.

File: csharp/src/Ice/StringUtil.cs (hunk 2 above)
Comment (pepone) [reply csharp/src/Ice/StringUtil.cs:26]: there is `IndexOfAny` but it uses a char array instead of a string for the match

https://docs.microsoft.com/en-us/dotnet/api/system.string.indexofany?view=netcore-3.1#System_String_IndexOfAny_System_Char___

File: csharp/src/Ice/StringUtil.cs [hunk 3]
```diff
@@ -602,16 +602,12 @@ public static string UnescapeString(string s, int start, int end, string special
             return l.ToArray();
         }
 
-        public static string JoinString(string[] values, string delimiters)
+        public static string JoinStringProperty(IReadOnlyList<string> values)
```
Comment (externl) [inline csharp/src/Ice/StringUtil.cs:605]: I think this should still have a comment.

File: csharp/src/Ice/StringUtil.cs [hunk 4]
```diff
@@ -602,16 +602,12 @@ public static string UnescapeString(string s, int start, int end, string special
...
-            if (delimiters.Length == 0)
-            {
-                throw new ArgumentException("delimiters cannot be empty", nameof(delimiters));
-            }
-
+            string delimiters = ", \n\r\t";
             char quote = '"';
             var result = new StringBuilder();
-            char delimiter = delimiters[0];
+            char delimiter = ' ';
```
Comment (externl) [inline csharp/src/Ice/StringUtil.cs:610]: In your previous commit this was `delimiters[0]`, which was `,`. Is this right?

Comment (pepone) [reply csharp/src/Ice/StringUtil.cs:610]: Yes, we should pick one either comma or whitespace

File: csharp/src/Ice/StringUtil.cs [hunk 5]
```diff
@@ -602,13 +602,60 @@ public static string UnescapeString(string s, int start, int end, string special
             return l.ToArray();
         }
 
+        public static string JoinStringProperty(IReadOnlyList<string> values)
+        {
+            string delimiters = ", \n\r\t";
```
Comment (externl) [inline csharp/src/Ice/StringUtil.cs:607]: Could we make this an array now? I think it would make space more obvious " ". All we do is iterate over it anyway.

Comment (pepone) [reply csharp/src/Ice/StringUtil.cs]: Yes fixed

File: csharp/src/Ice/StringUtil.cs (hunk 3 above)
Comment (pepone) [reply csharp/src/Ice/StringUtil.cs:610]: Added a comment

**Comment by bernardnormier** [comment]
Should we use an System.Collections.Generic.IEnumerable<string> parameter like string.Join?
Then, for the method name, maybe `StringUtil.ToPropertyValue`?

**Comment by externl** [comment]
> Then, for the method name, maybe `StringUtil.ToPropertyValue`?

I like `StringUtil.ToPropertyValue`.

File: csharp/src/Ice/StringUtil.cs [hunk 6]
```diff
@@ -602,13 +602,55 @@ public static string UnescapeString(string s, int start, int end, string special
             return l.ToArray();
         }
 
+        /// <summary>Concatenates a collection of strings in a format that is compatible with
+        /// <see cref="Communicator.GetPropertyAsList(string)"/>.</summary>
+        /// <param name="values">The collection that contains of strings to concatenate.</param>
```
Comment (bernardnormier) [inline csharp/src/Ice/StringUtil.cs:607]: Should be simply "The collection of strings to concatenate".

File: csharp/src/Ice/StringUtil.cs (hunk 4 above)
Comment (bernardnormier) [reply csharp/src/Ice/StringUtil.cs:615]: My preference is comma, because it's the delimiter used when concatenating multiple options in a URI (typical example, alt-endpoint).
